src/main/scala/com/microsoft/hyperspace/index/sources/delta/DeltaLakeFileBasedSource.scala

```diff
@@ -140,10 +140,10 @@ class DeltaLakeFileBasedSource(private val spark: SparkSession) extends FileBase
    * @param location Partitioned data location.
    * @return basePath to read the given partitioned location.
    */
-  override def partitionBasePath(location: FileIndex): Option[String] = {
+  override def partitionBasePath(location: FileIndex): Option[Option[String]] = {
     location match {
       case d: TahoeLogFileIndex =>
```

Comment:
Could you add `if (location.partitionSchema.nonEmpty)` here? and return `Some(None)` if `isEmpty` (same for DefaultFileBasedSource)

And could you remove `extractBasePath` function in RuleUtils.scala accordingly? Thanks!

Reply:
thanks @sezruby , added